rust/private/rustc.bzl

```diff
@@ -290,6 +290,12 @@ def collect_inputs(
 
     linker_depset = cc_toolchain.all_files
 
+    additional_transitive_inputs = depset([
+        additional_input
+        for linker_input in dep_info.transitive_noncrates.to_list()
```

Comment:
This is introducing quadratic behavior, transitive noncrates are now flattened on each rust_library transitively depending on the cc library.

Because of https://github.com/bazelbuild/bazel/issues/13132, and the current design of CcInfo, we cannot really avoid flattening the depset at some points. WDYT about:

* Flattening and constructing "expanded" additional_inputs depset (and propagating it through CrateInfos) when we collect_deps from CcInfo
* or only add additional inputs when you know this is a linking-like action (so if the only output is rlib, we don't add them)

Reply:
Hmm linkopts are a bit relevant here, if we don't add additional inputs, we cannot add linkopts as well.

Reply:
Updated to only add the additional inputs for linking-like actions (crate_type `bin`).